### ISSUE TITLE
DPO loss

### ISSUE THREAD

**Issue by JiuhaiChen** [huggingface/alignment-handbook#38]
I am training DPO with lora, the loss has weird behavior: will decrease sharply at the beginning of each epoch. I wonder if you have same issue before? 
<img width="548" alt="Screenshot 2023-11-17 at 12 28 28" src="https://github.com/huggingface/alignment-handbook/assets/27043988/ae677fef-a0a5-4098-b4e1-8ecd11de2983">

**Comment by ChenDRAG** [comment]
It seems that full finetuning has this problem, while lora doesn't. Could you share the training configuration?  Also how many GPUs are you using?

![image](https://github.com/huggingface/alignment-handbook/assets/40993476/184babce-d75c-420c-808d-6ced6cbb765b)

**Comment by JiuhaiChen** [comment]
Thanks for your reply. I don't try the full model fine-tuning. For the lora, i only changed: gradient_accumulation_steps: 1, per_device_train_batch_size: 16, per_device_eval_batch_size: 4, save_strategy: "epoch". I am using the 8 A6000.  Also, i am not sure if you observed the eval loss is increasing in the training. 
![image](https://github.com/huggingface/alignment-handbook/assets/27043988/59aadd91-9fe0-4c5e-a5e6-b90b56097136)

**Comment by JiuhaiChen** [comment]
Yeah, i agree eval loss does not matter.  For the lora, how many cards you are using?

**Comment by ChenDRAG** [comment]
8 A40 cards. My new experiments also encounter this problem.
![image](https://github.com/huggingface/alignment-handbook/assets/40993476/46b95d22-4919-49a8-80d0-8d6befb6ad77)

Difference between the two configurations
previous

bath size  4   accumulation 2 cards 8  lr  1e-7

new 
batch size 8  accumulation 1 cards 8  lr  1e-4

**Comment by NickWithBotronics** [comment]
I’m currently training a lora across all mistral modules with the standard setting with the exception of no eval, and a single batch size on a 3090. My loss is hitting .29 and it’s only been training for 180 steps. (.4 epochs)

**Comment by fblgit** [comment]
quite weird, i just trained the DPO and my loss is normal across epochs, pretty much similar to the results shared on hf model card.
how about rebase and try again ? definitively .29 or lower is because the model is seeing the right prediction token somehow.

**Comment by qychen2001** [comment]
In most cases, DPO will only train one epoch, and more epochs will cause a performance crash.
At the same time, smaller learning rates generally lead to better results, and it is recommended that you can start trying from 5e-7.
